src/crimson/os/seastore/onode_manager/staged-fltree/node.cc

```diff
@@ -1501,22 +1521,31 @@ eagain_future<Ref<Node>> InternalNode::get_or_track_child(
     context_t c, const search_position_t& position, laddr_t child_addr)
 {
   LOG_PREFIX(OTree::InternalNode::get_or_track_child);
-  bool level_tail = position.is_end();
-  Ref<Node> child;
   auto found = tracked_child_nodes.find(position);
   Ref<Node> this_ref = this;
-  return (found == tracked_child_nodes.end()
-    ? (Node::load(c, child_addr, level_tail
-       ).safe_then([this, position, c, FNAME] (auto child) {
-         child->as_child(position, this);
-         TRACET("loaded child untracked {} at pos({})",
-                c.t, child->get_name(), position);
-         return child;
-       }))
-    : (TRACET("loaded child tracked {} at pos({})",
-              c.t, found->second->get_name(), position),
-       eagain_ertr::make_ready_future<Ref<Node>>(found->second))
-  ).safe_then([this_ref, this, position, child_addr] (auto child) {
+  return seastar::now().then([this, found, position, child_addr, c, FNAME] {
+    if (found == tracked_child_nodes.end()) {
```

Comment:
if the continuation after `.then()` can be evaluated immediately, how about put something like:
```c++
return [&] {
  if (auto found = tracked_child_nodes.find(position);
      found != tracked_child_nodes.end()) {
    // early return
    return eagain_ertr::make_ready_future<Ref<Node>>(found->second);
  }
  // loading
  return // ...
}().safe_then([this_ref, this, position, child_addr] (auto child) {
  // ...
});
```

for two reasons:

1. less continuations. even the first one can "swallow" the second one.
2. less indent by early return. as the reader of the code can focus on the "normal" code path.

Reply:
Both done, thanks.